From 2b2f2767851eccb12e591c7a3fa432e6bf9db8f2 Mon Sep 17 00:00:00 2001
From: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
Date: Sat, 21 Feb 2015 23:10:35 -0500
Subject: [PATCH 39/45] curl-shim: clean up varargs

* keyserver/curl-shim.c (curl_easy_setopt) : ensure that va_end is
  called.

--

stdarg(3) says:
      Each invocation of va_start() must be matched by a
      corresponding invocation of va_end() in the same function.

Observed by Joshua Rogers <honey@internot.info>

Debian-Bug-Id: #773475

[dkg: rebased to STABLE-BRANCH-1-4]
Signed-off-by: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
---
 keyserver/curl-shim.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/keyserver/curl-shim.c b/keyserver/curl-shim.c
index ce510cb..72c0f04 100644
--- a/keyserver/curl-shim.c
+++ b/keyserver/curl-shim.c
@@ -155,6 +155,8 @@ curl_easy_setopt(CURL *curl,CURLoption option,...)
       break;
     }
 
+  va_end(ap);
+
   return handle_error(curl,CURLE_OK,NULL);
 }
 
-- 
2.1.4

